### PR TITLE
feat: add ::Selenium::WebDriver::Support::AbstractEventListener support as Appium::Support::EventFiringBridge

### PR DESCRIPTION
**Pull request by KazuCocoa**
Part of https://github.com/appium/ruby_lib_core/issues/455

Potentially this is sufficient. I haven't tested yet.

### CODE REVIEW

File: lib/appium_lib_core/common/base/driver.rb [hunk 1]
```diff
@@ -44,16 +44,18 @@ class Driver < ::Selenium::WebDriver::Driver
...
 
-        private
-
         def initialize(bridge: nil, listener: nil, **opts)
           # For ::Appium::Core::Waitable
           @wait_timeout = opts.delete(:wait_timeout)
           @wait_interval = opts.delete(:wait_interval)
 
-          super
+          @bridge = listener ? ::Appium::Core::Base::AppiumEventFiringBridge.new(bridge, listener) : bridge
```
Comment (tayrawr) [inline lib/appium_lib_core/common/base/driver.rb:52]: I have a concern around this line which might potentially impact execution. I believe we'll need to create an Appium bridge (if one isn't passed in) to dispatch commands across before initializing the AppiumEventFiringBridge (similar to how it's being handled in Selenium's webdriver).

https://github.com/SeleniumHQ/selenium/blob/trunk/rb/lib/selenium/webdriver/common/driver.rb#L74

Could we possibly do the same within the Appium driver init, or do you foresee any issues that might be caused from it?

https://github.com/appium/ruby_lib_core/blob/master/lib/appium_lib_core/common/base/driver.rb#L61-L83

Comment (KazuCocoa) [reply lib/appium_lib_core/common/base/driver.rb:52]: I'll write test code and check, but yes, current thinking is the bridge will be the appium's one.

**Comment by KazuCocoa** [comment]
I think almost done.

@tayrawr Could you try out this branch?

As the test code, a subclass of `::Selenium::WebDriver::Support::AbstractEventListener.new` as `listener` may work via `EventFiringBridge` which inherits Selenium's one.

**Comment by tayrawr** [comment]
@KazuCocoa, from my tests so far, everything seems great! I was able to create a listener and trigger its event methods.

**Comment by KazuCocoa** [comment]
FYI: v6.3.0 has this change

**Pull request merged**